### ISSUE TITLE
Unexpected Style/HashSyntax consistency offence

### ISSUE THREAD

**Issue by sunquan1991** [rubocop-hq/rubocop#4229]
I don't know whether this is what it's supposed to be.

I got a **"Use hash rockets syntax."** wanrning from code
```ruby
scope :module => :users do
  resources :registrations, only: %i(create update destroy)
end
``` 
with `UseHashRocketsWithSymbolValues: true`. 

I think RuboCop checked the syntax consistency of `only: [:create, :update, :destroy]` and `scope :module => :users`, which is however not what I expected.

--------

## Expected behavior

No warning

## Actual behavior

Got a **"Use hash rockets syntax."** wanrning

## Steps to reproduce the problem

Have the following in rubocop.yml
```yml
Style/HashSyntax:
  EnforcedStyle: ruby19
  UseHashRocketsWithSymbolValues: true
  PreferHashRocketsForNonAlnumEndingSymbols: false
```
Run rubocop against following:
```ruby
scope :module => :users do
  resources :registrations, only: %i(create update destroy)
end
```

## RuboCop version

```
$ rubocop -V
0.47.1 (using Parser 2.4.0.0, running on ruby 2.4.1 x86_64-darwin16)
```

**Comment by ashmaroli** [comment]
probably because
```ruby
resources :blogs, only: [:create, :update, :destroy]
```
is treated  by the Ruby interpreter as
```ruby
resources :blogs, {:only => [:create, :update, :destroy]}
```

**Comment by retroalgic** [comment]
@ashmaroli Sorry, I didn't get your point. According to https://github.com/bbatsov/rubocop/blob/022e7322b731b06bd31ce5ef1bac378b27551ed0/config/default.yml#L631
it force hashes that **have a symbol value** to use hash rockets
```ruby
resources :blogs, only: [:create, :update, :destroy]
``` 
and 
```ruby
resources :blogs, {:only => [:create, :update, :destroy]}
```
should both be fine as the value is an array rather than symbol. And I don't think it has something to do with interpreter.

**Comment by ashmaroli** [comment]
> Sorry, I didn't get your point.

@retroalgic My bad. I myself don't know what I was thinking when I commented that..
Will be deleting that and this comment shortly..

**Comment by ashmaroli** [comment]
Couldn't reproduce this issue on
```
0.48.0 (using Parser 2.4.0.0, running on ruby 2.3.1 i386-mingw32)
```
though.

**Comment by bbatsov** [comment]
@ashmaroli did you use the same config options as the OP?

**Comment by ashmaroli** [comment]
## Setup:
```
.
|__ lib/
|    |__ test.rb
|__ Gemfile
|__ .rubocop.yml
```
```ruby
# lib/test.rb

resources :blogs, only: [:create, :update, :destroy]

```
```ruby
# Gemfile

source 'https://rubygems.org'
gem 'rubocop', require: false
```
```yaml
# .rubocop.yml

---
Style/HashSyntax:
  EnforcedStyle: ruby19
  UseHashRocketsWithSymbolValues: true
  PreferHashRocketsForNonAlnumEndingSymbols: false
```
## Test Run
```sh
bundle exec rubocop -V
# => 0.48.0 (using Parser 2.4.0.0, running on ruby 2.3.1 i386-mingw32)
```
```sh
bundle exec rubocop
# =>
# Inspecting 2 files
# .C
#
# Offenses:
#
# lib/test.rb:1:25: C: Use %i or %I for an array of symbols.
# resources :blogs, only: [:create, :update, :destroy]
#                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
#
# 2 files inspected, 1 offense detected
```

**Comment by retroalgic** [comment]
@ashmaroli @bbatsov Sorry, my fault.
The code is in a block actually:
```ruby
scope :module => :users do
  resources :registrations, only: [:create, :update, :destroy]
end
```
I think RuboCop checked the consistency with `scope :module => :users`. But it's not what I expected. Also for me, the problem can be solved by enabling `Style/SymbolArray`. Thx!

**Comment by retroalgic** [comment]
I also got the same offence with the following
```ruby
class ApplicationController < ActionController::Base
  # For APIs, you may want to use :null_session instead.
  protect_from_forgery :with => :exception

  def default_url_options
    { locale: I18n.locale.downcase }
  end
end
```

**Comment by jonas054** [comment]
When `UseHashRocketsWithSymbolValues` is `true`, the hash rocket syntax should be used when there is at least one symbol literal among the values of a hash. But the `HashSyntax` cop caches the `@force_hash_rockets` flag and uses it for all subsequent hash literals in the file. This is a bug.
